linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -39,17 +40,26 @@ private[emitter] final class KnowledgeGuardian(config: Emitter.Config) {
    *  maintaining it. It is a better trade-off to invalidate everything in
    *  the rare events where they do change.
    */
-  def update(linkingUnit: LinkingUnit): Boolean = {
-    val hasInlineableInit = computeHasInlineableInit(linkingUnit)
-    val staticFieldMirrors =
-      computeStaticFieldMirrors(linkingUnit.topLevelExports)
+  def update(moduleSet: ModuleSet): Boolean = {
+    val hasInlineableInit = computeHasInlineableInit(moduleSet)
+    val staticFieldMirrors = computeStaticFieldMirrors(moduleSet)
 
-    var objectClass: LinkedClass = null
+    // Object is optional, because the module splitter might remove everything.
+    var objectClass: Option[LinkedClass] = None
```

Comment:
Have you considered short-cutting this situation right at the start of `Emitter.emit()`, before calling `emitInternal()`? That way we wouldn't have to deal with this corner case everywhere else in the emitter.

Reply:
Hmmm... I can, but this is the only place that needs changing IIUC. So I'm not sure short-cutting at the beginning is simpler (or clearer). I do not feel strongly about it though. So if you prefer that, I can do that.

Reply:
I don't really have a strong preference either. We can keep it as is.